guardrails/async_guard.py

```diff
@@ -0,0 +1,1353 @@
+import asyncio
+import contextvars
+import inspect
+import json
+import os
+import warnings
+from copy import deepcopy
+from string import Template
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    Union,
+    cast,
+    overload,
+)
+
+from guardrails_api_client.models import AnyObject
+from guardrails_api_client.models import Guard as GuardModel
+from guardrails_api_client.models import (
+    History,
+    HistoryEvent,
+    ValidatePayload,
+    ValidationOutput,
+)
+from guardrails_api_client.types import UNSET
+from langchain_core.messages import BaseMessage
+from langchain_core.runnables import Runnable, RunnableConfig
+from pydantic import BaseModel
+from pydantic.version import VERSION as PYDANTIC_VERSION
+from typing_extensions import deprecated
+
+from guardrails.api_client import GuardrailsApiClient
+from guardrails.classes import OT, InputType, ValidationOutcome
+from guardrails.classes.credentials import Credentials
+from guardrails.classes.generic import Stack
+from guardrails.classes.history import Call
+from guardrails.classes.history.call_inputs import CallInputs
+from guardrails.classes.history.inputs import Inputs
+from guardrails.classes.history.iteration import Iteration
+from guardrails.classes.history.outputs import Outputs
+from guardrails.errors import ValidationError
+from guardrails.llm_providers import (
+    get_async_llm_ask,
+    get_llm_api_enum,
+    get_llm_ask,
+    model_is_supported_server_side,
+)
+from guardrails.logger import logger, set_scope
+from guardrails.prompt import Instructions, Prompt
+from guardrails.rail import Rail
+from guardrails.run import AsyncRunner
+from guardrails.schema import Schema, StringSchema
+from guardrails.stores.context import (
+    Tracer,
+    get_call_kwarg,
+    get_tracer_context,
+    set_call_kwargs,
+    set_tracer,
+    set_tracer_context,
+)
+from guardrails.utils.hub_telemetry_utils import HubTelemetry
+from guardrails.utils.llm_response import LLMResponse
+from guardrails.utils.reask_utils import FieldReAsk
+from guardrails.utils.validator_utils import get_validator
+from guardrails.validator_base import FailResult, Validator
+
+
+class AsyncGuard(Runnable, Generic[OT]):
```

Comment:
This should probably just inherit from the original Guard class and overwrite `__call__` and `parse`, that way we're not reimplementing everything else that doesn't change

Reply:
done